test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -570,17 +570,31 @@ public void DefaultContentRootIsApplicationBasePath()
         }
 
         [Fact]
-        public void DefaultApplicationNameWithNoStartupThrows()
+        public void DefaultWebHostBuilderWithNoStartupThrows()
         {
-            var builder = new ConfigurationBuilder();
             var host = new WebHostBuilder()
                 .UseServer(new TestServer());
 
-            var ex = Assert.Throws<ArgumentException>(() => host.Build());
+            var ex = Assert.Throws<InvalidOperationException>(() => host.Build());
+
+            Assert.Equal("No service for type 'Microsoft.AspNetCore.Hosting.IStartup' has been registered.", ex.Message);
```

Comment:
I tried to create a better error message but didn't come up with anything nice. The best I was able to come up with was 
```c#
throw new InvalidOperationException($"No startup methods configured. Please specify startup methods via WebHostBuilder.UseStartup, WebHostBuilder.Configure, injecting {nameof(IStartup)} or specifying the {nameof(WebHostOptions.StartupAssembly)}.");
```
Which is not concise either.